src/generator/AutoRest.AzureResourceSchema/ResourceMarkdownGenerator.cs

```diff
@@ -436,5 +450,46 @@ private JToken JsonSchemaToJTokenImpl(string propName, JsonSchema jsonSchema, St
                     return new JValue(val);
             }
         }
+
+        /// <summary>
+        /// Flattens properties in the AllOf and Properties sections into one list.
+        /// </summary>
+        /// <param name="jsonSchema">The JSON schema.</param>
+        /// <returns>A dictionary of all possible properties.</returns>
+        private IDictionary<string, JsonSchema> FlattenProperties(JsonSchema jsonSchema)
+        {
+            var merged = new Dictionary<string, JsonSchema>();
+            FlattenPropertiesImpl(merged, new[] { jsonSchema });
+
+            return merged;
+        }
+
+        /// <summary>
+        /// Recursively flattens properties in the AllOf and Properties sections into one list (don't call this one call FlattenProperties).
+        /// </summary>
+        /// <param name="merged">The dictionary to contain the merged content.</param>
+        /// <param name="items">The items to merge into the dictionary.</param>
+        private void FlattenPropertiesImpl(Dictionary<string, JsonSchema> merged, IEnumerable<JsonSchema> items)
+        {
+            foreach (var item in items)
+            {
+                if (item.AllOf != null)
+                {
+                    FlattenPropertiesImpl(merged, item.AllOf);
+                }
+                else if (item.Properties != null)
```

Comment:
It's not clear to me that each item will have one of AllOf, Properties, or Ref to flatten. Are there any cases where multiple of these clauses would need to be hit?